### PR TITLE
Don't restore duplicated dependencies

### PR DESCRIPTION
**Pull request by yufeih**
[AB#230952](https://dev.azure.com/ceapex/Engineering/_workitems/edit/230952/)

Further optimize multiple docsets restore to avoid re-fetching the same repo multiple times in a single build.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6036)

### CODE REVIEW

File: src/docfx/restore/FileResolver.cs [hunk 1]
```diff
@@ -83,7 +87,7 @@ public string ResolveFilePath(SourceInfo<string> file)
 
         public async Task Download(SourceInfo<string> file)
         {
-            if (!UrlUtility.IsHttp(file))
+            if (!UrlUtility.IsHttp(file) || s_downloadedUrls.Contains(file))
```
Comment (live1206) [inline src/docfx/restore/FileResolver.cs:90]: There is still chance to download more than once, right?

Comment (yufeih) [reply src/docfx/restore/FileResolver.cs:90]: yes, tiny.

Comment (live1206) [reply src/docfx/restore/FileResolver.cs:90]: ok, fair trade off to achieve better perf without lock

**Pull request merged**